components/org.apache.stratos.python.cli/src/main/python/cli/CLI.py

```diff
@@ -0,0 +1,1530 @@
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_activate_tenant(self, tenant_domain, opts=None):
+        """Activate a tenant."""
+        try:
+            if not tenant_domain:
+                print("usage: activate-tenant <TENANT_DOMAIN> ")
+            else:
+                activate_tenant = Stratos.activate_tenant(tenant_domain)
+                if activate_tenant:
+                    print("You have successfully activated the tenant : "+tenant_domain)
+                else:
+                    print("Could not activate tenant : "+tenant_domain)
+        except BadResponseError as e:
+            self.perror(str(e))
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_deactivate_tenant(self, tenant_domain, opts=None):
+        """Deactivate a tenant."""
+        try:
+            if not tenant_domain:
+                print("usage: deactivate-tenant <TENANT_DOMAIN> ")
+            else:
+                activate_tenant = Stratos.deactivate_tenant(tenant_domain)
+                if activate_tenant:
+                    print("You have successfully deactivated the tenant : "+tenant_domain)
+                else:
+                    print("Could not deactivate tenant : "+tenant_domain)
+        except BadResponseError as e:
+            self.perror(str(e))
+
+    """
+    # Cartridges
+     * list-cartridges
+     * list-cartridges-by-filter
+     * describe-cartridge
+     * add-cartridge
+     * remove-cartridge
+
+    """
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_list_cartridges(self, line , opts=None):
+        """Retrieve details of available cartridges."""
+        try:
+            cartridges = Stratos.list_cartridges()
+            table = PrintableTable()
+            rows = [["Type", "Category", "Name", "Description", "Version", "Multi-Tenant"]]
+            for cartridge in cartridges:
+                rows.append([cartridge['type'], cartridge['category'], cartridge['displayName'], cartridge['description'],
+                             cartridge['version'], "True" if cartridge['multiTenant'] == 1 else "False"])
+            table.add_rows(rows)
+            table.print_table()
+        except BadResponseError as e:
+            self.perror(str(e))
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_list_cartridges_by_filter(self, filter_text , opts=None):
+        """Retrieve details of available cartridges."""
```

Comment:
Might be better to auto-generate method comments and include parameter descriptions.